packages/shared/lib/core/layer-2/actions/loadEvmAddressForSelectedAccount.ts

```diff
@@ -0,0 +1,16 @@
+import { getSelectedAccount } from '@core/account/stores'
+import { Ledger } from '@core/ledger/classes'
+import { getActiveProfile } from '@core/profile'
+import { COIN_TYPE } from '@core/network/constants'
+
+export async function loadEvmAddressForSelectedAccount(): Promise<void> {
+    try {
+        const { evmAddress, index } = getSelectedAccount() ?? {}
+        if (!evmAddress) {
+            const coinType = COIN_TYPE[getActiveProfile()?.network?.id] ?? 1
+            await Ledger.generateEvmAddress(coinType, index ?? 0, false)
```

Comment:
i feel like we wanna cache this, because the account can be accessed without a ledger connection. 
If we login like that, we wouldnt be able to see our addresses

Reply:
Yup, but we would cache this ideally together with the `depositAddress`. Will be in a future PR.